### PR TITLE
Improve Swift usage in platform_channel_swift some more

### PR DESCRIPTION
**Pull request by jamesderlin**
Make a bunch of changes that willlarche requested in
https://github.com/flutter/flutter/pull/21712:

* Use `static let` within `enum` namespaces for String constants.
* Use `guard` for early exits and to avoid forced unwrapping.
* Explicitly qualify a selector.
* Fix a retain cycle by using a weak reference.

### CODE REVIEW

File: examples/platform_channel_swift/ios/Runner/AppDelegate.swift [hunk 1]
```diff
@@ -13,19 +27,21 @@ import Flutter
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
     GeneratedPluginRegistrant.register(with: self)
-    let controller: FlutterViewController = window?.rootViewController as! FlutterViewController
-    let batteryChannel = FlutterMethodChannel(name: "samples.flutter.io/battery",
+    guard let controller = window?.rootViewController as? FlutterViewController else {
+      return false
```
Comment (willlarche) [inline examples/platform_channel_swift/ios/Runner/AppDelegate.swift]: Returning false here is something you'd do if you're doing deep linking and don't wanna handle the URL that's passed. In this case, if the view controller isn't what you want, you should `fatalError("rootViewController is not type FlutterViewController")`

[Examples of using fatalError in IOSched-iOS](https://github.com/google/iosched-ios/search?q=fatal&unscoped_q=fatal) if you need to be reassured.

File: examples/platform_channel_swift/ios/Runner/AppDelegate.swift [hunk 2]
```diff
@@ -61,22 +77,21 @@ import Flutter
   }
 
   private func sendBatteryStateEvent() {
```
Comment (willlarche) [inline examples/platform_channel_swift/ios/Runner/AppDelegate.swift]: This looks so good this way!

**Review by willlarche — APPROVED** [review]
Love it!

File: examples/platform_channel_swift/ios/Runner/AppDelegate.swift (hunk 1 above)
Comment (jamesderlin) [reply examples/platform_channel_swift/ios/Runner/AppDelegate.swift]: Done.

**Review by willlarche — APPROVED** [review]
Thank you!

**Pull request merged**